software/base/src/main/java/org/apache/brooklyn/entity/software/base/lifecycle/MachineLifecycleEffectorTasks.java

```diff
@@ -708,7 +710,21 @@ protected void doStop(ConfigBag parameters, Callable<StopMachineDetails<Integer>
 
         DynamicTasks.queue("pre-stop", new PreStopCustomTask());
 
+        // BROOKLYN-263:
+        // With this change the stop effector will always wait for machinelocation to provision, so it can stop it.
+        // However in some situations if the stop effector is called in less than 8 seconds after start was called,
+        // it may be safe to skip this wait and proceed with stop.
+        // Hesitant about parameterising this.
         Maybe<MachineLocation> machine = Machines.findUniqueMachineLocation(entity().getLocations());
+        if (machine.isAbsent()) {
```

Comment:
This feels dangerous, for if start were to fail and not get a machine - we'd wait a long time, presumably.